pkg/api/v1/types.go

```diff
@@ -116,6 +119,7 @@ type VMList struct {
 
 // VMSpec is a description of a VM
 type VMSpec struct {
+	// Domain is the actual libvirt domain in JSON format. When creating a VM, it is converted to XML.
```

Comment:
For the second point: it's bound to happen soon. It's just easier to navigate the code with usable godoc for the time being.